docs/rfcs/0007-synchronized-releases.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: page
+title: Synchronizing the branching and versioning of Android Components with the Mozilla release trains
+permalink: /rfc/0007-synchronized-releases
+---
+
+* Start date: 2021-03-22
+* RFC PR: [#9973](https://github.com/mozilla-mobile/android-components/pull/9973)
+
+## Summary
+
+Synchronizing the branching and versioning of Android Components with the Mozilla release trains (GeckoView) to simplify release processes and integration by:
+
+* Having just one component implementing `concept-engine` with GeckoView
+* Having the `master` branch track only GeckoView Nightly
+* Creating branches and releases tracking matching GeckoView releases (e.g. GeckoView 89.0 -> Android Components 89.0)
+
+## Motivation
+
+Initially we introduced three `concept-engine` implementations for GeckoView, one for each GeckoView release channel:
+
+* `browser-engine-gecko-nightly`
+* `browser-engine-gecko-beta`
+* `browser-engine-gecko`
+
+ This allowed us to release Android Components independently from GeckoView and consumers could use the latest AC release regardless of which GeckoView channel they wanted to use it with.
+
+This flexibility comes at a cost and makes our release process and maintenance more difficult. For every new GeckoView release across all channels (nightly, beta, release) we want to ship a new release of Fenix (and with that Android Components). So there is already an implicit synchronization of releases, but it is manual and confusing. By making this dependency explicit through synchronized releases we can simplify our processes.
```

Comment:
maybe we can add some examples on why this makes maintenance more difficult? or what is this cost

Reply:
This is a good point. I will try to clarify that.